packages/grid/x-data-grid-premium/src/DataGridPremium/DataGridPremium.tsx

```diff
@@ -26,9 +30,36 @@ const DataGridPremiumRaw = React.forwardRef(function DataGridPremium<R extends G
 ) {
   const props = useDataGridPremiumProps(inProps);
   const privateApiRef = useDataGridPremiumComponent(props.apiRef, props);
+  const headerFiltersRef = React.useRef<HTMLDivElement>(null);
 
+  privateApiRef.current.register('private', {
+    headerFiltersElementRef: headerFiltersRef,
+  });
```

Comment:
Maybe move it to `useGridColumnHeaders` or `GridBody` to avoid duplication in `DataGridPro` and `DataGridPremium`?

Reply:
Done ✅